### DISCUSSION TITLE
Saving a Pay::Charge to a Devise User

### DISCUSSION THREAD

**Discussion by Sumner-David** [pay-rails/pay#515] [Q&A] [answered]
*Using Gem V2 as that is what the tutorial used

I've been following the [Stripe Checkout video](https://www.youtube.com/watch?v=MKWq_soCzsQ&t=454s) for a site that uses a devise model named Subscribers. 

When a payment is successful i get redirected as expected but the Pay::Charge model does not get created when looking in the rails console. Would this be because my subscriber has no set first_name and last_name and Pay somehow uses those to create the object? 

The same happens when I make a successful subscription

**Comment by excid3** [comment, marked as answer]
Make sure you're running the Stripe CLI so that webhooks can record charges and subscription changes.

**Reply by Sumner-David** [reply]
Double checked my credentials and you were right!
Was debugging for ages and it turned out i spelled 'signing' in the STRIPE_SIGNING-SECRET wrong 😬

Additionally, is there a way to set the product name when using the checkouts? It seems to use the default name set in the configs. 
I did a workaround where it checks the `stripe processor plan` with a switch statement but Pay might already have a way

**Reply by excid3** [reply]
Stripe Checkout will display the name of the Product ID you give it.

**Reply by Sumner-David** [reply]
Sorry, I meant on the section where you used `current_user.payment_processor.checkout` and adding a name to that plan. 

Doing 
```
@checkout_session = current_user.payment_processor.checkout(
      mode: 'subscription',
      line_items: 'price_1KF4flJqnqKeEzRUt97sqI3C',
      name: 'Named subscription'
    )
```
Comes back with `Received unknown parameter: name` and when omitting the `name` it just uses the default name setting. I've been looking around the code in the V2 version and can't seem to find a where it is being defined so far in my search
